<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>MainWindow</class>
 <widget class="QMainWindow" name="MainWindow">
  <property name="windowModality">
   <enum>Qt::NonModal</enum>
  </property>
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>686</width>
    <height>524</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>MainWindow</string>
  </property>
  <widget class="QWidget" name="CentralWidget">
   <widget class="QLabel" name="ProgramTitle">
    <property name="geometry">
     <rect>
      <x>10</x>
      <y>10</y>
      <width>251</width>
      <height>21</height>
     </rect>
    </property>
    <property name="text">
     <string>Vigenère cipher encoder/decoder</string>
    </property>
   </widget>
   <widget class="QPushButton" name="CopyButton">
    <property name="geometry">
     <rect>
      <x>10</x>
      <y>380</y>
      <width>141</width>
      <height>28</height>
     </rect>
    </property>
    <property name="statusTip">
     <string>Copy encoded text to clipboard</string>
    </property>
    <property name="text">
     <string>Copy to clipboard</string>
    </property>
   </widget>
   <widget class="QPushButton" name="SaveButton">
    <property name="geometry">
     <rect>
      <x>160</x>
      <y>380</y>
      <width>111</width>
      <height>28</height>
     </rect>
    </property>
    <property name="statusTip">
     <string>Save encoded text to file</string>
    </property>
    <property name="text">
     <string>Save as file...</string>
    </property>
   </widget>
   <widget class="QLabel" name="CreditsLabel1">
    <property name="geometry">
     <rect>
      <x>10</x>
      <y>440</y>
      <width>661</width>
      <height>26</height>
     </rect>
    </property>
    <property name="text">
     <string>Copyright 2023</string>
    </property>
   </widget>
   <widget class="QLabel" name="CreditsLabel0">
    <property name="geometry">
     <rect>
      <x>10</x>
      <y>420</y>
      <width>201</width>
      <height>20</height>
     </rect>
    </property>
    <property name="text">
     <string>Program written by Katsumi</string>
    </property>
   </widget>
   <widget class="QLabel" name="AuthorLinkLabel">
    <property name="geometry">
     <rect>
      <x>10</x>
      <y>460</y>
      <width>261</width>
      <height>26</height>
     </rect>
    </property>
    <property name="text">
     <string>https://github.com/KatsumiKougen</string>
    </property>
   </widget>
   <widget class="QWidget" name="layoutWidget">
    <property name="geometry">
     <rect>
      <x>10</x>
      <y>30</y>
      <width>664</width>
      <height>341</height>
     </rect>
    </property>
    <layout class="QVBoxLayout" name="MainLayout">
     <item>
      <layout class="QVBoxLayout" name="PlainTextInputLayout">
       <item>
        <layout class="QGridLayout" name="VgVariantGridLayout">
         <item row="0" column="2">
          <widget class="QRadioButton" name="VgVariant1OptionRadio">
           <property name="cursor">
            <cursorShape>PointingHandCursor</cursorShape>
           </property>
           <property name="statusTip">
            <string>Use Vigenère-RC4 cipher (8-bit) </string>
           </property>
           <property name="text">
            <string>Vigenère-RC4 (8-bit)</string>
           </property>
          </widget>
         </item>
         <item row="1" column="0">
          <widget class="QLabel" name="CryptogramModeLabel">
           <property name="text">
            <string>Mode</string>
           </property>
          </widget>
         </item>
         <item row="1" column="3">
          <widget class="QRadioButton" name="DecodeModeRadio">
           <property name="cursor">
            <cursorShape>PointingHandCursor</cursorShape>
           </property>
           <property name="text">
            <string>Decode</string>
           </property>
          </widget>
         </item>
         <item row="0" column="1">
          <widget class="QRadioButton" name="VgVariant0OptionRadio">
           <property name="cursor">
            <cursorShape>PointingHandCursor</cursorShape>
           </property>
           <property name="statusTip">
            <string>Use Vigenère cipher</string>
           </property>
           <property name="text">
            <string>Vigenère</string>
           </property>
          </widget>
         </item>
         <item row="1" column="2">
          <widget class="QRadioButton" name="EncodeModeRadio">
           <property name="cursor">
            <cursorShape>PointingHandCursor</cursorShape>
           </property>
           <property name="text">
            <string>Encode</string>
           </property>
          </widget>
         </item>
         <item row="0" column="3">
          <widget class="QRadioButton" name="VgVariant2OptionRadio">
           <property name="cursor">
            <cursorShape>PointingHandCursor</cursorShape>
           </property>
           <property name="statusTip">
            <string>Use Vigenère-RC4 cipher (4-bit)</string>
           </property>
           <property name="text">
            <string>Vigenère-RC4 (4-bit)</string>
           </property>
          </widget>
         </item>
         <item row="0" column="0">
          <widget class="QLabel" name="VgVariantLabel">
           <property name="text">
            <string>Variant</string>
           </property>
          </widget>
         </item>
        </layout>
       </item>
       <item>
        <widget class="QPlainTextEdit" name="UserInputEdit">
         <property name="font">
          <font>
           <family>Monospace</family>
          </font>
         </property>
         <property name="cursor" stdset="0">
          <cursorShape>IBeamCursor</cursorShape>
         </property>
         <property name="locale">
          <locale language="English" country="Philippines"/>
         </property>
         <property name="horizontalScrollBarPolicy">
          <enum>Qt::ScrollBarAlwaysOff</enum>
         </property>
        </widget>
       </item>
      </layout>
     </item>
     <item>
      <layout class="QHBoxLayout" name="KeywordInputLayout">
       <item>
        <widget class="QLabel" name="KeywordLabel">
         <property name="statusTip">
          <string>Type your keyword here</string>
         </property>
         <property name="text">
          <string>Keyword</string>
         </property>
        </widget>
       </item>
       <item>
        <widget class="QPlainTextEdit" name="KeywordInputEdit">
         <property name="enabled">
          <bool>true</bool>
         </property>
         <property name="sizePolicy">
          <sizepolicy hsizetype="Expanding" vsizetype="Preferred">
           <horstretch>0</horstretch>
           <verstretch>0</verstretch>
          </sizepolicy>
         </property>
         <property name="font">
          <font>
           <family>Monospace</family>
          </font>
         </property>
         <property name="cursor" stdset="0">
          <cursorShape>IBeamCursor</cursorShape>
         </property>
         <property name="statusTip">
          <string>Type your keyword here</string>
         </property>
         <property name="locale">
          <locale language="English" country="Philippines"/>
         </property>
         <property name="horizontalScrollBarPolicy">
          <enum>Qt::ScrollBarAlwaysOff</enum>
         </property>
        </widget>
       </item>
      </layout>
     </item>
     <item>
      <layout class="QVBoxLayout" name="LowerLayout">
       <item>
        <widget class="QLabel" name="OutputLabel">
         <property name="text">
          <string>Output</string>
         </property>
        </widget>
       </item>
       <item>
        <widget class="QPlainTextEdit" name="OutputSection">
         <property name="font">
          <font>
           <family>Monospace</family>
          </font>
         </property>
         <property name="cursor" stdset="0">
          <cursorShape>IBeamCursor</cursorShape>
         </property>
         <property name="locale">
          <locale language="English" country="Philippines"/>
         </property>
         <property name="frameShape">
          <enum>QFrame::StyledPanel</enum>
         </property>
         <property name="frameShadow">
          <enum>QFrame::Sunken</enum>
         </property>
         <property name="horizontalScrollBarPolicy">
          <enum>Qt::ScrollBarAlwaysOff</enum>
         </property>
         <property name="readOnly">
          <bool>true</bool>
         </property>
         <property name="textInteractionFlags">
          <set>Qt::NoTextInteraction</set>
         </property>
         <property name="placeholderText">
          <string/>
         </property>
        </widget>
       </item>
      </layout>
     </item>
    </layout>
   </widget>
  </widget>
  <widget class="QStatusBar" name="ProgramStatusBar"/>
 </widget>
 <tabstops>
  <tabstop>VgVariant0OptionRadio</tabstop>
  <tabstop>VgVariant1OptionRadio</tabstop>
  <tabstop>VgVariant2OptionRadio</tabstop>
  <tabstop>EncodeModeRadio</tabstop>
  <tabstop>DecodeModeRadio</tabstop>
  <tabstop>UserInputEdit</tabstop>
  <tabstop>KeywordInputEdit</tabstop>
  <tabstop>OutputSection</tabstop>
  <tabstop>CopyButton</tabstop>
  <tabstop>SaveButton</tabstop>
 </tabstops>
 <resources/>
 <connections/>
</ui>
